src/gui/guiConfirmRegistration.cpp

```diff
@@ -96,8 +96,34 @@ void GUIConfirmRegistration::regenerateGui(v2u32 screensize)
 				"Join to confirm account creation or click Cancel to "
 				"abort.");
 		char info_text_buf[1024];
+
+#ifdef _MSC_VER
+		int sizeBuffer = sizeof(info_text_buf) -
+				 (info_text_template.length() - 8 + address.length() +
+						 m_playername.length());
+
+		if (sizeBuffer >= 0) {
+			sprintf(info_text_buf, info_text_template.c_str(),
+					address.c_str(), m_playername.c_str());
+		} else {
+			warningstream << "Failed to generate info_text. Buffer is too "
+					 "small"
+				      << std::endl;
```

Comment:
Since this should never happen in practice, IMO it's okay (and much simpler) to abort the dialog or not show any text at all

Reply:
at least the player can still play

Reply:
Realistically, when is length(server name) + length(player name) + some fixes amount around 300 ever going to exceed 1024?

Instead of trying to do the right thing in weird cases, it's better to fail.

Reply:
ok i do that